### PR TITLE
Makes overlay IPv6 validation checks optional

### PR DESCRIPTION
**Pull request by GoelDeepak**
## High-level description
As of DC/OS 1.11, IPv6 overlay network is optional. Hence, the
validation check shouldn't assert on the presence of IPv6 overlay.
This patch makes modifies the validation checks such that IPv6
overlay could be optional

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS-21095](https://jira.mesosphere.com/browse/DCOS-21095) Turn IPv6 Overlay Off in IPv4 Environment

## Checklist for all PRs

  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by GoelDeepak** [comment]
@asridharan  @urbanserj  request for review

**Comment by GoelDeepak** [comment]
@mesosphere-mergebot bump-ee

File: gen/calc.py [hunk 1]
```diff
@@ -305,15 +305,14 @@ def validate_dcos_overlay_network(dcos_overlay_network):
...
-    assert 'vtep_subnet6' in overlay_network.keys(), (
-        'Missing "vtep_subnet6" in overlay configuration {}'.format(overlay_network))
-
-    try:
-        ipaddress.ip_network(overlay_network['vtep_subnet6'])
-    except ValueError as ex:
-        raise AssertionError(
-            "Incorrect value for vtep_subnet6: {}."
-            " Only IPv6 values are allowed".format(overlay_network['vtep_subnet6'])) from ex
+    # IPv6 is optional
```
Comment (asridharan) [inline gen/calc.py]: @GoelDeepak this makes sense. The only thing that bothers me if some specifies an IPv6 virtual network and doesn't specify `vtep_subnet6` things will break badly?

Comment (GoelDeepak) [reply gen/calc.py]: I don't know how to handle that. Any suggestions?

Comment (asridharan) [reply gen/calc.py]: We could walk through all the overlay networks present in `dcos_overlay_network` and look for the `ip6` subnet key. If any of the virtual networks has this key than `vtep_subnet6` becomes required?

**Comment by GoelDeepak** [comment]
@mesosphere-mergebot bump-ee

**Comment by GoelDeepak** [comment]
@mesosphere-mergebot Ready For Review

**Comment by urbanserj** [comment]
@mesosphere-mergebot label [Ready For Review]

**Pull request merged**